metaflow/plugins/argo/argo_workflows.py

```diff
@@ -846,6 +846,8 @@ def _compile_workflow_template(self):
                 .templates(self._container_templates())
                 # Exit hook template(s)
                 .templates(self._exit_hook_templates())
+                # Sidecar templates (Daemon Containers)
+                .templates(self._daemon_container_templates())
```

Comment:
Can we make this optional through the cli - —with-daemon-container/—no-with-daemon-container ? That way folks can opt out if needed

Reply:
done. One thought on the wording for the toggle though: Currently we only run heartbeats, and in the future some other run-scope-processes in the default container.

what happens with the toggle though, if we introduce a way for users to launch their own daemon containers alongside tasks? should these also not deploy if `--no-with-daemon-container` is specified, or should it not apply to user defined ones?